### PR TITLE
delete helmrelease when deleting ns

### LINKED ISSUE #5122: When the namespace xxx is deleted, the helmreleases in the namespace xxx  is not deleted
**Describe the Bug**
When the namespace xxx is deleted, the helmreleases in the namespace xxx  is not deleted. This will cause a problem: when curling http://ip:port/kapis/openpitrix.io/v1/workspaces/{workspace}/clusters/{cluster}/applications，you will get the helmrelease in the xxx namespace which has been deleted.

image

The namesapce su-text has been deleted. I think we shouldn't show the helmreleae in the su-text namesapce.

**Versions Used**
KubeSphere:3.2.1
Kubernetes: 1.21

**Environment**
How many nodes and their hardware configuration:

CentOS 7.6 / 3 masters:  8cpu/8g; 3 nodes: 8cpu/16g

### PR DESCRIPTION
**Pull request by suwliang3**
Signed-off-by: suwanliang_yewu <suwanliang_yewu@cmss.chinamobile.com>

### What type of PR is this?
/kind bug

### What this PR does / why we need it:
When the namespace xxx is deleted, the helmreleases in the namespace xxx is not deleted. This will cause a problem: when curling [http://ip:port/kapis/openpitrix.io/v1/workspaces/{workspace}/clusters/{cluster}/applications，you](http://ip:port/kapis/openpitrix.io/v1/workspaces/%7Bworkspace%7D/clusters/%7Bcluster%7D/applications%EF%BC%8Cyou) will get the helmrelease in the xxx namespace which has been deleted.

### Which issue(s) this PR fixes:

Fixes #5122 

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?
None

```release-note
fix of a previous Known Issue
```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/models/tenant/tenant.go [hunk 1]
```diff
@@ -372,6 +380,28 @@ func (t *tenantOperator) DeleteNamespace(workspace, namespace string) error {
 	return t.k8sclient.CoreV1().Namespaces().Delete(context.Background(), namespace, *metav1.NewDeleteOptions(0))
 }
 
+func (t *tenantOperator) DeleteRls(namespace string) error {
+
+	ls := map[string]string{}
+	ls[constants.NamespaceLabelKey] = namespace
```
Comment (xyz-li) [inline pkg/models/tenant/tenant.go:386]: You must also add this key `constants.WorkspaceLabelKey` to the map.

**Comment by xyz-li** [comment]
I think you need to commit this pr to master branch.

**Comment by suwliang3** [comment]
> I think you need to commit this pr to master branch.

**Comment by suwliang3** [comment]
@xyz-li @wansir Is there anyone dealing with this pr now?

**Comment by wansir** [comment]
@suwliang3  The resource recycling logic should move to the namespace controller.

https://github.com/kubesphere/kubesphere/blob/d3cf418c6eaa4b55b983ca3315dbf75b4d0e42ac/pkg/controller/namespace/namespace_controller.go#L220-L237

**Comment by suwliang3** [comment]
@wansir The resource recycling logic has moved to the namespace controller.

**Comment by suwliang3** [comment]
> @wansir The resource recycling logic has moved to the namespace controller.

/cc @xyz-li @wansir

**Comment by suwliang3** [comment]
@xyz-li @wansir Is there anyone dealing with this pr now?

**Comment by suwliang3** [comment]
> @xyz-li @wansir Is there anyone dealing with this pr now?

@wansir @xyz-li what is the situation now?

**Pull request closed without merging**